README.md

```diff
@@ -96,8 +96,24 @@ $httpServer = new HttpServer($socket, $callback);
 
 #### Return type of the callback function
 
-The return type of the callback function **must** be a response object.
-Other types aren't allowed and will lead to an 'HTTP 500 Internal Server Error' for the client.
+The return type of the callback function **must** be a [response object](https://packagist.org/packages/ringcentral/psr7) or
+a [promise](https://github.com/reactphp/promise).
+
+For heavy calculations you should consider using promises. Not using them can slow down the server.
+
+```php
+$callback = function (Request $request) {
+    return new Promise(function ($resolve, $reject) use ($request) {
+        // Some heavy calculations
+        $resolve(new Response());
```

Comment:
This appears a bit unclear: What is "heavy" here, why can not using promises "slow down" the server?